apps/src/lib/levelbuilder/unit-editor/UnitEditor.jsx

```diff
@@ -706,13 +726,14 @@ class UnitEditor extends React.Component {
                   </p>
                 </HelpTip>
               </label>
-              {this.props.hasCourse && (
-                <div>
-                  <p>
-                    This unit is part of a course. Go to the course edit page to
-                    publish the course and its units.
-                  </p>
-                  {/*
+              {this.props.hasCourse &&
+                this.state.publishedState !== PublishedState.in_development && (
```

Comment:
I get where you're going with this, but it also seems like it adds to the list of things that needs to be done in a short time period when preparing for launch. if I want to be able to hide a unit within a script, why not let me do it ahead of time, while the unit is still in development? the tooltip already provides a great explanation of what this will affect.

this isn't a "hard no" on this idea -- I'm just trying to think through what the actual user experience will be like for content editors and whether this actually helps them. curious what your thoughts are here.

Reply:
I did this because we already prevent you from setting the unit published state individually if you are setting it to the same value as the course published state. So it seemed weird to have the checkbox in that case.

Reply:
ok, this seems like a reasonable compromise then.